### ISSUE TITLE
Selected value in SearchChoice.single not working in Mobile but works fine in Web

### ISSUE THREAD

**Issue by agustin-garcia** [lcuis/search_choices#115]
Hello,
When User selects a value from a SearchChoice.single widget, the value goes back to hint value
This behavior only happens when in physical mobile devices and mobile emulators (Android), because if I test in web mode, the widget behaves exactly as expected, meaning that it changes the original hint to User's selected value.

I looked into the debugger option, but no error/warning is being thrown there.

I'm using:
search_choices 2.2.5
Android Studio Flamingo | 2022.2.1
Flutter 3.10.6
Dart 3.0.6
environment:
  sdk: ">=2.18.0 <4.0.0"

NOTE: I'm using the widget inside a GetX observable as shown in the code snippet below

...
RxString selectedOwner = ''.obs;
...
Obx(
  () => SearchChoices.single(
    items: usersController.usersDropDownItems,
    value: selectedOwner.value,
    label: "Responsible",
    hint: "Responsible",
    searchHint: "Select responsible",
    onChanged: (value) {
      selectedOwner.value = value;
    },
    onClear: () {
      selectedOwner.value = '';
    },
    dialogBox: true,
    isExpanded: true,
  )

),

**Comment by lcuis** [comment]
Hi @agustin-garcia !

Thanks for raising this.

I am on holiday and don't have the hardware to investigate.

Somehow, your issue reminds me of #111 .

It is likely due to the way your items are built and the way the values are passed both ways.

Can you show how you built your list of items please?

**Comment by praveen0308** [comment]
I am having a similar kind of issue:
I have two single-selection dialogs. I want them to have an initial value. When I am intializing it with String data type it's working fine but trying with custom data objects its not reflecting the initial value. 
Code:

`
  CategoryModel selectedCategory = CategoryModel();
  var categories = List<DropdownMenuItem<CategoryModel>>.empty(growable: true);


// Initializing categories dropdown
for (var element in widget.categories) {
      categories.add(DropdownMenuItem(value: element,child: Text(element.name ?? ""),));
    }

    selectedCategory = CategoryModel(id: content.categoryId,name: content.category);


// Setting into the SearchChoice.single

SearchChoices.single(
                  displayClearIcon: false,
                  underline: "",
                  padding: const EdgeInsets.all(5),
                  items: categories,
                  value: selectedCategory,
                  hint: "Select one",
                  searchHint: "Select one",
                  onChanged: (value) {
                    setState(() {
                      selectedCategory = value;
                    });
                  },
                  isExpanded: true,
                )

`

**Comment by lcuis** [comment]
Hi @praveen0308 ,

Thanks for sharing this. I have a few questions:
1) Does `CategoryModel` have a `.toString()` method?
2) Is the `selectedCategory ` in the `categories` list?

**Comment by praveen0308** [comment]
Hi @lcuis ,
Thanks for asking very logical questions. I had `.toString()` method in CategoryModel and `selectedCategoory` in` categories` list. But after some investigation I found `selectedCategory` was not considered in the list or it was not considering `selectedCategory` identical with same `category` in `categories`. I implemented `Equatable` approach. It solved and it's working smoothly.

Thanks again!
